### PR TITLE
Patch Sample and Field endpoints

### PR DESCRIPTION
**Pull request by CamronStaley**
## What changes are proposed in this pull request?

Adding endpoints to patch at the sample or field level following these guidelines: https://datatracker.ietf.org/doc/html/rfc6902

## How is this patch tested? If it is not, please explain why.

Tested locally with unit tests and calling directly to the endpoint, then verifying to state of the sample in Mongo after edits are made.

### Is this a user-facing change that should be mentioned in the release notes?

-   [x] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [x] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

* **New Features**
  * Support JSON Patch (RFC 6902) for editing samples via application/json-patch+json.
  * Endpoint to patch individual items inside list fields by item ID.

* **Improvements**
  * Unified JSON-to-model transformations with clearer "No transform registered for class ..." errors.
  * Stricter content-type handling (415) and consistent 404s for missing datasets/samples.
  * Aggregated validation error reporting for PATCH requests.

* **Tests**
  * Expanded unit tests covering JSON Patch parsing, patch methods, route behavior, and edge cases.

### CODE REVIEW

File: fiftyone/server/routes/sample.py [hunk 1]
```diff
@@ -53,49 +104,30 @@ async def patch(self, request: Request, data: dict) -> dict:
...
+            result = self._handle_patch(sample, data)
+        elif content_type == "application/json-patch+json":
+            result = handle_json_patch(sample, data)
+        else:
             raise HTTPException(
-                status_code=404,
-                detail=f"Sample '{sample_id}' not found in dataset '{dataset_id}'",
+                status_code=415,
+                detail=f"Unsupported Content-Type '{content_type}'",
             )
```
[the remark this answers is not in the document]
Comment (tom-vx51) [reply fiftyone/server/routes/sample.py:119]: +1 to provide better compatibility with third-party clients

File: fiftyone/server/utils/jsonpatch/methods.py [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+    src: T, path: Union[str, jsonpointer.JsonPointer], value: V
+) -> Union[T, V]:
+    """The "add" operation performs one of the following functions,
+    depending upon what the target location references:
+
+    o  If the target location specifies an array index, a new value is
+        inserted into the array at the specified index.
+    o  If the target location specifies an object member that does not
+        already exist, a new member is added to the object.
+    o  If the target location specifies an object member that does exist,
```
Comment (tom-vx51) [inline fiftyone/server/utils/jsonpatch/methods.py:90]: just curious, is this allowed under json patch spec? or just convenient to allow

File: fiftyone/server/utils/jsonpatch/methods.py [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+
+    pointer = to_json_pointer(path)
+    if not pointer.parts:
+        return value
+
+    target = get(src, jsonpointer.JsonPointer.from_parts(pointer.parts[:-1]))
+    name = pointer.parts[-1]
+
+    try:
+        if hasattr(target, "__getitem__"):
```
Comment (tom-vx51) [inline fiftyone/server/utils/jsonpatch/methods.py]: [nit] shared logic with `get`

Comment (j053y) [reply fiftyone/server/utils/jsonpatch/methods.py]: Yep you're right, that particular value should actually be `__setitem__`. 

If you're talking more about the logic: get, set, and del all follow a similar pattern (of checking what the source object can do), but are just different enough that I thought abstracting it would make it more confusing and harder to work on.

File: fiftyone/server/utils/jsonpatch/methods.py (hunk 2 above)
Comment (j053y) [reply fiftyone/server/utils/jsonpatch/methods.py:90]: The docstrings for each operation are directly copied from the JSONPatch spec. So anything there is valid.
